packages/wrangler/src/zones.ts

```diff
@@ -22,25 +22,23 @@ export interface Zone {
  * ```
  * However, in the case of patterns that _can't_ be parsed as a hostname
  * (primarily the pattern `*/ /*`), we fall back to the `zone_name`
- * (and in the absence of that throw an error).
+ * (and in the absence of that return undefined).
  * @param route
  */
 export function getHostFromRoute(route: Route): string | undefined {
+	let host: string | undefined;
+
 	if (typeof route === "string") {
-		return getHostFromUrl(route);
+		host = getHostFromUrl(route);
 	} else if (typeof route === "object") {
-		try {
-			return getHostFromUrl(route.pattern);
-		} catch (e) {
-			if (
-				(e as { code: string }).code === "ERR_INVALID_URL" &&
-				"zone_name" in route
-			) {
-				return getHostFromUrl(route.zone_name);
-			}
-			throw e;
+		host = getHostFromUrl(route.pattern);
+
+		if (host === undefined && "zone_name" in route) {
+			host = getHostFromUrl(route.zone_name);
 		}
 	}
+
+	return host;
 }
```

Comment:
This exception handling logic was inconsistent (not covering all calls to getHostFromUrl) and therefore confusing (took me a while to realise it wasn't a mistake). So now we try our best inside getHostFromUrl to create a valid url string and, if it still is not, instead of relying on exceptions, return `undefined` to indicate we could not infer a host and to fallback to trying `route.zone_name`.